### PR TITLE
Add const getters to ParticleBox

### PR DESCRIPTION
**Pull request by FelixTUD**
Const added to getters in ParticleBox.hpp for compatibility with new ISAAC Renderer

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
@FelixTUD  Could you please accept the invitation as PIConGPU developer and than force push this PR again. The CI will only check if your changes if you are a developer.

File: include/pmacc/particles/memory/boxes/ParticlesBox.hpp [hunk 1]
```diff
@@ -330,7 +330,7 @@ class ParticlesBox : protected DataBox<PitchedBox<SuperCell<T_Frame>, DIM> >
         return false;
     }
 
-    HDINLINE SuperCellType& getSuperCell( DataSpace<DIM> idx )
+    HDINLINE SuperCellType& getSuperCell( DataSpace<DIM> idx ) const
```
Comment (sbastrakov) [inline include/pmacc/particles/memory/boxes/ParticlesBox.hpp]: This method is technically constant, but seems to be logically not constant, as it returns a reference to some internal stuff. In my mind, maybe a better solution is to leave the old method as is and add a const twin, which returns a const reference and then use that new one?

I am also not sure about some others changed, maybe the similar argument applies.

Comment (ax3l) [reply include/pmacc/particles/memory/boxes/ParticlesBox.hpp]: If it compiles it's valid C++ and `const` on a method technically only means the method causes no internal side effects.
I think it's ok for now to progress on what he is implementing.

**Comment by psychocoderHPC** [comment]
@FelixTUD What is the state of this PR.

**Comment by FelixTUD** [comment]
> 
> 
> @FelixTUD What is the state of this PR.

@psychocoderHPC 
I haven't tried the new changes proposed by @sbastrakov yet. But @ax3l seems to be fine with this pull request in the current state. Should I still try to change the consts in the logical sense next week?

**Comment by sbastrakov** [comment]
@FelixTUD I think my suggestions are not critical (although stylistically I'm for it). If you feel it is better to delay it or not implement it at all, please feel free to.

**Comment by psychocoderHPC** [comment]
discussed offline: we will keep this PR for now because there are some problems if we like to transform it so that const methods only return const values.

**Pull request merged**